main.tf

```diff
@@ -73,3 +73,40 @@ resource "aws_elasticache_replication_group" "this" {
 
   tags = var.tags
 }
+
+resource "awscc_elasticache_serverless_cache" "this" {
+  count = var.enabled && var.use_serverless ? 1 : 0
+
+  serverless_cache_name = var.name
+  description           = "${var.name} ElastiCache Redis Serverless"
+  engine                = "redis"
+  major_engine_version  = var.engine_version
+
+  cache_usage_limits = {
+    data_storage = {
+      maximum = var.max_data_storage
+      unit    = "GB"
+    }
+    ecpu_per_second = {
+      maximum = var.max_ecpu_per_second
+    }
+  }
+
+  user_group_id = var.serverless_user_group_id
+
+  final_snapshot_name = "${var.name}-elasticache-serverless-final-snapshot"
+  kms_key_id          = var.kms_key_id
+  security_group_ids  = var.security_groups
+  subnet_ids          = var.subnets
+
+  daily_snapshot_time      = var.daily_snapshot_time
+  snapshot_arns_to_restore = var.snapshot_arns_to_restore
+  snapshot_retention_limit = var.snapshot_retention_limit
+
+  tags = toset([
```

Comment:
@paul-ylz For tags, we normally define it in this format `map(string)`:
```terraform
tags = {
  "app-name" = "Somersault"
  "owner"        = "Jane Doe"
}
```

The `awscc_elasticache_serverless_cache` tag attribute is set to accept an array of key-value pair https://registry.terraform.io/providers/hashicorp/awscc/latest/docs/resources/elasticache_serverless_cache#nestedatt--tags so we'll have to do this.

This will not be necessary once we have the aws provider resource for elasticache serverless since the tags will be defined in the provider.

Reply:
I just realised that I misunderstood your question. Actually, there's no need to use toset function here. I'll remove it.